File: 프로그래머스/lv1/70128. 내적/내적.js
// 네비게이터
// 반복문을 사용하여 a[i]와 b[i] 값을 곱하여 더한 값을 answer 에 push 합니다.



// ----------
function solution(a, b) {
  var answer = 0;
  
  for (let i=0; i<a.length; i++) {
    answer += a[i]*b[i]
  }

  return answer;
}

console.log(solution([1,2,3,4],[-3,-1,0,2]))

File: 프로그래머스/1/70128. 내적/내적.py
# def solution(a, b):
#     # return sum([i*j for i in a for j in b]) # 이중포문이기 때문에 x
#     # return sum([lambda key=i,j: for i in a for j in b]) # 문법오류
#     sum = 0
#     for k in range(len(a)):
#         sum += a[k]*b[k]
#     return sum

# def solution(a,b):
#     return sum([x*y for x,y in zip(a,b)])

# solution = lambda x,y: sum(x*y for x,y in zip(a,b)) -> 에러

solution = lambda a, b: sum(x*y for x, y in zip(a,b))